services/libssh/cve_2018_10933_server.py

```diff
@@ -0,0 +1,323 @@
+                if self.completion_event is not None:
+                    self.completion_event.set()
+                if self.auth_handler is not None:
+                    self.auth_handler.abort()
+                for event in self.channel_events.values():
+                    event.set()
+                try:
+                    self.lock.acquire()
+                    self.server_accept_cv.notify()
+                finally:
+                    self.lock.release()
+            self.sock.close()
+        except:
+            # Don't raise spurious 'NoneType has no attribute X' errors when we
+            # wake up during interpreter shutdown. Or rather -- raise
+            # everything *if* sys.modules (used as a convenient sentinel)
+            # appears to still exist.
+            if self.sys.modules is not None:
+                raise
+
+
+class ParamikoSSHServer(paramiko.ServerInterface):
+
+    def __init__(self):
+        self.event = threading.Event()
+
+    def check_channel_request(self, kind, chanid):
+        if kind == "session":
+            return paramiko.OPEN_SUCCEEDED
+        return paramiko.OPEN_FAILED_ADMINISTRATIVELY_PROHIBITED
+
+    def check_auth_password(self, username, password):
+        data = {
+            EVENT_TYPE_FIELD_NAME: SSH_ALERT_TYPE,
+            USERNAME_FIELD_NAME: username,
+            PASSWORD_FIELD_NAME: password
+        }
+        self.alert(self.socket, **data)
+        return paramiko.AUTH_FAILED
+
+    def check_auth_publickey(self, username, key):
+        data = {
+            EVENT_TYPE_FIELD_NAME: SSH_ALERT_TYPE,
+            USERNAME_FIELD_NAME: username,
+            ADDITIONAL_FIELDS_FIELD_NAME: {
+                KEY_FIELD_NAME: key.get_base64()
+            }
+        }
+        self.alert(self.socket, **data)
+        return paramiko.AUTH_FAILED
+
+    def check_auth_gssapi_with_mic(
+        self, username, gss_authenticated=paramiko.AUTH_FAILED, cc_file=None
+    ):
+        return paramiko.AUTH_FAILED
+
+    def check_auth_gssapi_keyex(
+        self, username, gss_authenticated=paramiko.AUTH_FAILED, cc_file=None
+    ):
+        return paramiko.AUTH_FAILED
+
+    def enable_auth_gssapi(self):
+        return True
+
+    def get_allowed_auths(self, username):
+        return "gssapi-keyex,gssapi-with-mic,password,publickey"
+
+    def check_channel_shell_request(self, channel):
+        self.event.set()
+        return True
+
+    def check_channel_pty_request(
+        self, channel, term, width, height, pixelwidth, pixelheight, modes
+    ):
+        return True
+
+
+class SSHRequestHandler(StreamRequestHandler):
+    alert = None
+    chan = None
+    transport = None
+    paramiko_server = None
+
+    def handle(self):
+        self.transport = CVETransport(self.connection, gss_kex=True)
+        self.transport.alert = self.alert
+        self.transport.set_gss_host(socket.getfqdn(""))
+        try:
+            self.transport.load_server_moduli()
+        except:
```

Comment:
Why add except: raise? I realize you probably used this for debugging, but do you still need this code here?